### DISCUSSION TITLE
Expire

### DISCUSSION THREAD

**Discussion by Ischafak** [nuxt-modules/turnstile#261] [Q&A]
### 📚 Is your documentation request related to a problem?

How can i get new token after it expire normalls as i know it has expired-callback event

**Comment by dargmuesli** [comment]
The README states: 

> To manually regenerate the token, @nuxtjs/turnstile exposes the reset function directly via a [template ref](https://vuejs.org/guide/essentials/template-refs.html).

And gives the following example:

```vue
<template>
  <NuxtTurnstile ref="turnstile" />
  <button @click="turnstile.reset()">Reset token in template</button>
  <button @click="reset()">Reset token from handler</button>
</template>

<script setup>
  // you can call this template ref anything
  const turnstile = ref()

  function reset() {
    turnstile.value?.reset()
  }
</script>
```
